api/rtx/src/unsupported_malloc.c

```diff
@@ -0,0 +1,162 @@
+    /* adjust the heap size to the new heap start address */
+    heap_size = (uint32_t) __StackLimit - heap_start;
+
+    /* page heap now extends from the previous main heap start address
+     * to the new main heap start address */
+    extern uint32_t __uvisor_page_size;
+    page_allocator_init(__end__, (void *) heap_start, &__uvisor_page_size);
+    box_index_init((void *) heap_start, heap_size);
+}
+
+
+int uvisor_page_malloc(UvisorPageTable *const table)
+{
+    return page_allocator_malloc(table);
+}
+
+int uvisor_page_free(const UvisorPageTable *const table)
+{
+    return page_allocator_free(table);
+}
+
+/* This is the fallback implementation for using the page allocator from uVisor
+ * inside an OS as a normal function.
+ * Be aware that the page allocator is not re-entrant, so the OS must provide a
+ * mutex implementation to enable thread-safety!
+ */
+#define DPRINTF(...) {}
+#define g_active_box 0
+#define vmpu_is_box_id_valid(...) 0
+#define vmpu_public_flash_addr(...) 1
+#define vmpu_sram_addr(...) 1
+#define HALT_ERROR(id, ...) {}
+
+/* Implement mutex for page allocator. */
+static osMutexId g_page_allocator_mutex_id = NULL;
+static int32_t g_page_allocator_mutex_data[4];
+static const osMutexDef_t g_page_allocator_mutex = { g_page_allocator_mutex_data };
+
+static void page_allocator_mutex_aquire()
+{
+    if (g_page_allocator_mutex_id == NULL) {
+        /* Create mutex if not already done. */
+        g_page_allocator_mutex_id = osMutexCreate(&g_page_allocator_mutex);
+        if (g_page_allocator_mutex_id == NULL) {
+            /* Mutex failed to be created. */
+            return;
+        }
+    }
+
+    osMutexWait(g_page_allocator_mutex_id, osWaitForever);
+}
+
+#define UVISOR_PAGE_ALLOCATOR_MUTEX_AQUIRE  page_allocator_mutex_aquire()
+#define UVISOR_PAGE_ALLOCATOR_MUTEX_RELEASE osMutexRelease(g_page_allocator_mutex_id)
+
+/* Alignment of MPU regions is not required anymore, however we still require
+ * a 32B alignment, to have some page size granularity. */
+static inline int vmpu_is_region_size_valid(uint32_t size)
+{
+    return ((size & ~31) == size);
+}
+static inline uint32_t vmpu_round_up_region(uint32_t addr, uint32_t size)
+{
+    if (!vmpu_is_region_size_valid(size)) {
+        return 0;
+    }
+    const uint32_t mask = size - 1;
+    /* Adding the mask can overflow. */
+    const uint32_t rounded_addr = addr + mask;
+    /* Check for overflow. */
+    if (rounded_addr < addr) {
+        /* This means the address was too large to align. */
+        return 0;
+    }
+    /* Mask the rounded address to get the aligned address. */
+    return (rounded_addr & ~mask);
+}
+static inline uint32_t page_table_read(uint32_t addr)
+{
+    return *((uint32_t *) addr);
+}
+static inline void page_table_write(uint32_t addr, uint32_t data)
+{
+    *((uint32_t *) addr) = data;
+}
+
+/* Include the original page allocator source directly. */
+#include "source/page_allocator.c_inc"
```

Comment:
> I understand why you did this but it makes unicorns cry.

😭 

> Would it still be ok if you instead save this `rtx_malloc_wrapper.c` file as a .h file and then include it conditionally in `page_allocator.c`?

But then I have the same, just the other way around. It's not better, just because it's a `.h` file.
I'd still have to copy the `page_allocator.c` file over from uVisor core. but this way, no unsupported code is in uVisor core.